### PR TITLE
LPD-24689 Get the id of the actual creator user, instead of the current user

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPD-24689](https://liferay.atlassian.net/browse/LPD-24689)

### CODE REVIEW

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 1]
```diff
@@ -747,6 +747,81 @@ public void testSendNotificationWithRoles() throws Exception {
 			notificationTemplate2);
 	}
 
+	@Test
+	public void testSendNotificationWithSystemObject() throws Exception {
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:751]: @devpedrotavares rename it to `testSendNotificationWithSystemObjectDefinition`

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 2]
```diff
@@ -747,6 +747,81 @@ public void testSendNotificationWithRoles() throws Exception {
...
+			RandomTestUtil.randomString(),
+			ObjectActionExecutorConstants.KEY_NOTIFICATION,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_UPDATE,
+			UnicodePropertiesBuilder.put(
+				"notificationTemplateId",
+				notificationTemplate.getNotificationTemplateId()
+			).build(),
+			false);
+
+		User accountAuthorUser = UserTestUtil.addOmniadminUser();
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:797]: @devpedrotavares rename it to `omniadminUser` or just `user`

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java [hunk 3]
```diff
@@ -203,14 +203,13 @@ private static Map<String, Object> _getVariables(
...
 		Map<String, Object> allowedVariables =
 			HashMapBuilder.<String, Object>put(
 				"creator",
 				() -> {
 					if (objectDefinition.isUnmodifiableSystemObject()) {
-						return userId;
+						return null;
 					}
 
 					return MapUtil.getString(objectEntry, "userId");
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java:215]: @devpedrotavares please inline if possible:

```
if (objectDefinition.isUnmodifiableSystemObject()) {
    return null;
}

return MapUtil.getString((Map<String, Object>)payloadJSONObject.get("objectEntry"), "userId");
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java (hunk 3 above)
Comment (devpedrotavares) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java:215]: The `objectEntry` is used in another place as well

**Comment by devpedrotavares** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:report:2566560232

**Comment by devpedrotavares** [comment]
ci:report:2566562975

**Comment by devpedrotavares** [comment]
ci:report:2566562975

**Comment by devpedrotavares** [comment]
ci:forward:force

**Pull request closed without merging**